### PR TITLE
Issue #3412311 by Kingdutch: Remove update hooks for Open Social 11 and earlier

### PR DESCRIPTION
**Pull request by Kingdutch**
⚠️ This PR is part of a set of stacked commits please do not rebase this PR but instead ping @Kingdutch 

## Problem
We have a lot of older update hooks which should've already been executed. As per the <a href="https://www.drupal.org/docs/drupal-distributions/open-social/updating">documentation for updating Open Social</a> major versions may not be skipped, so we should be able to rely on people moving properly from 11.x to 12.x and running all update hooks currently in 11.x, only receiving new update hooks in 12.x.

This means that for 13.x we know that only the 12.x update hooks might be missing from people's databases and older update hooks can be removed.

## Solution
Remove all update hooks that are present in Open Social 11 and earlier for the Open Social 13 release.

## Issue tracker
https://www.drupal.org/project/social/issues/3412311
https://getopensocial.atlassian.net/browse/PROD-27898

## How to test
- [ ] Update from 12.0.0 to 13.0.0 should work

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Release notes
All update hooks prior to Open Social 11 have been cleaned up. Updates should not skip major versions. Follow the [update documentation](https://www.drupal.org/docs/drupal-distributions/open-social/updating) and update to the latest version of Open Social 12 before upgrading to Open Social 13.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by Kingdutch** [comment]
The follow-users test that tested an upgrade path from a fixture is now broken because we've removed those update hooks. That means that we also no longer need the test (since the code under test no longer exists).

image

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3678-7mouvnln1079gv0uhbd3sqs1hqasgth5.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6596b04215a27ec46ac321b6

File: modules/custom/social_lets_connect/social_lets_connect.install [hunk 1]
```diff
@@ -6,15 +6,8 @@
  */
 
 /**
- * Uninstall social_lets_connect_usage module.
+ * Implements hook_update_last_removed().
  */
-function social_core_update_10001(): void {
```
Comment (ronaldtebrake) [inline modules/custom/social_lets_connect/social_lets_connect.install:11]: Nice copy paste here before 😂

**Review by ronaldtebrake — APPROVED** [review]

**Pull request merged**